db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2200 @@
+       *    should shorten the range that A applies over. That is, a source is succeeded before in time by a more valid
+       *    source.
+       *
+       *    A:     ++++++++
+       *    B:  +++++++
+       *        BBBBBBBAAAA
+       */
+      @Test
+      void testForwardOverlap() throws SQLException {
+        // create our sources and their per-window events
+        ExternalSource a = new ExternalSource(
+            "A",
+            SOURCE_TYPE,
+            DERIVATION_GROUP,
+            "2024-01-01T00:00:00Z",
+            "2024-01-01T00:30:00Z",
+            "2024-01-01T02:00:00Z",
+            CREATED_AT
+        );
+        ExternalEvent aE = new ExternalEvent(
+            a.key() + "_event",
+            SOURCE_TYPE,
+            a.key(),
+            DERIVATION_GROUP,
+            "2024-01-01T1:10:00Z",
+            DURATION
+        );
+        ExternalSource b = new ExternalSource(
+            "B",
+            SOURCE_TYPE,
+            DERIVATION_GROUP,
+            "2024-01-02T00:00:00Z",
+            "2024-01-01T00:00:00Z",
+            "2024-01-01T01:00:00Z",
+            CREATED_AT
+        );
+        ExternalEvent bE = new ExternalEvent(b.key() + "_event", SOURCE_TYPE, b.key(),
+                                             DERIVATION_GROUP, b.start_time(), DURATION);
+
+        // verify the ranges are as expected
+        // insert sources
+        insertExternalSource(a);
+        insertExternalSource(b);
+
+        // insert events
+        insertExternalEvent(aE);
+        insertExternalEvent(bE);
+
+        var results = getDerivedEvents();
+
+        // both ranges should only have a single element and be fully present
+        final List<DerivedEvent> expectedResults = List.of(
+            new DerivedEvent(
+                "A_event",
+                SOURCE_TYPE,
+                "A",
+                DERIVATION_GROUP,
+                "2024-01-01 01:10:00+00",
+                "00:00:00.000001",
+                "{[\"2024-01-01 01:00:00+00\",\"2024-01-01 02:00:00+00\")}",
+                "2024-01-01 00:00:00+00"
+            ),
+            new DerivedEvent(
+                "B_event",
+                SOURCE_TYPE,
+                "B",
+                DERIVATION_GROUP,
+                "2024-01-01 00:00:00+00",
+                "00:00:00.000001",
+                "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 01:00:00+00\")}",
+                "2024-01-02 00:00:00+00"
+            )
+        );
+        assertEquals(expectedResults.size(), results.size());
+        assertTrue(results.containsAll(expectedResults));
+      }
+
+      /**
+       * This test is an overlapping case wherein the source of higher precedence (the more recently valid one - B)
+       *    should shorten the range that A applies over. In this case, a source is succeeded after in time by a more
+       *    valid source.
+       *
+       *    A:  +++++++
+       *    B:     ++++++++
+       *        AAABBBBBBBB
```

Comment:
Test does not match the diagram. Test does the following overlap:
```
A:  ++++++
B:     +++
    AAABBB
```
Please adjust either diagram or test.

Reply:
I extended the length of B.